app/models/concerns/orchestration/dhcp.rb

```diff
@@ -164,8 +164,13 @@ def queue_dhcp_update
   # do we need to update our dhcp reservations
   def dhcp_update_required?
     # IP Address / name changed, or 'rebuild' action is triggered and DHCP record on the smart proxy is not present/identical.
-    return true if ((old.ip != ip) || (old.hostname != hostname) || provision_mac_addresses_changed? || (old.subnet != subnet) || (operatingsystem.boot_filename(old.host) != operatingsystem.boot_filename(self.host)) ||
-                    (!old.build? && build? && !all_dhcp_records_valid?))
+    return true if ((old.ip != ip) ||
+      (old.hostname != hostname) ||
+      provision_mac_addresses_changed? ||
+      (old.subnet != subnet) ||
+      (operatingsystem.boot_filename(old.host) != operatingsystem.boot_filename(self.host)) ||
```

Comment:
Below is code that checks if self is a host or not. If you're unconditionally using host here, then the old code below is probably no longer correct, right? Looking that the code that should be covered by the `boot_filename` you're introducing here anyway.

Additionally, should it be:
```suggestion
      (old.operatingsystem.boot_filename(old.host) != operatingsystem.boot_filename(self.host)) ||
```

Reply:
Yeah this could be a problem when OS changes.